lib/ansible/inventory/manager.py

```diff
@@ -223,7 +223,10 @@ def parse_source(self, source, cache=False):
         parsed = False
         display.debug(u'Examining possible inventory source: %s' % source)
 
+        # use binary for path functions
         b_source = to_bytes(source)
+        # source should already be 'text' but JIC, ensure it
+        source = to_text(source)
```

Comment:
We shouldn't do this here as one of the ideas of the unicode sandwich is to always have text unless explicitly using bytes.